librz/syscall/syscall.c

```diff
@@ -287,27 +402,21 @@ RZ_API RzList *rz_syscall_list(RzSyscall *s) {
 	return list;
 }
 
-/* io and sysregs */
-RZ_API const char *rz_syscall_get_io(RzSyscall *s, int ioport) {
+/**
+ * \brief Searches the sysregs SDB database inside librz/reg/d for the type (mmio/reg) and
+ * 		  returns its value as a string
+ *
+ * \param s reference to RzSyscall
+ * \param type reference to the type of key: mmio/reg
+ * \param port reference to the io port
+ */
+RZ_API const char *rz_sysreg_get(RzSyscall *s, const char *type, ut64 port) {
 	rz_return_val_if_fail(s, NULL);
-	int i;
-	const char *name = rz_syscall_sysreg(s, "io", ioport);
-	if (name) {
-		return name;
-	}
-	for (i = 0; s->sysport[i].name; i++) {
-		if (ioport == s->sysport[i].port) {
-			return s->sysport[i].name;
-		}
-	}
-	return NULL;
-}
-
-RZ_API const char *rz_syscall_sysreg(RzSyscall *s, const char *type, ut64 num) {
-	rz_return_val_if_fail(s, NULL);
-	if (!s->db) {
+	if (!s->srdb) {
 		return NULL;
 	}
-	const char *key = sdb_fmt("%s,%" PFMT64d, type, num);
-	return sdb_const_get(s->db, key, 0);
+	RzSysregItem *item = ht_up_find(s->srdb->port, port, NULL);
+	if (item)
+		return item->comment;
+	return NULL;
```

Comment:
I missed to compare the `type` here. What I mean is, right now the `comment` gets returned whether the type is mmio or io.

I'll fix it one of the upcoming patches.

Reply:
yes, remember that we are committing in dev, therefore is normal to do this.